balancer-js/src/modules/pools/factory/weighted/weighted.factory.spec.ts

```diff
@@ -76,15 +104,67 @@ describe('creating weighted pool', async () => {
           return isSameAddress(log.address, factoryAddress);
         })
         .map((log) => {
-          try {
-            return weightedPoolFactoryInterface.parseLog(log);
-          } catch (error) {
-            console.error(error);
-            return null;
-          }
+          return weightedPoolFactoryInterface.parseLog(log);
         })
         .find((parsedLog) => parsedLog?.name === 'PoolCreated');
+      if (poolCreationEvent) {
+        poolAddress = poolCreationEvent.args.pool;
+      }
       expect(!!poolCreationEvent).to.be.true;
+      return;
+    });
+    it('should init join a pool', async () => {
+      const signerAddress = await signer.getAddress();
+      const weightedPoolInterface = new Interface(WeightedPool__factory.abi);
+      const pool = new Contract(poolAddress, weightedPoolInterface, provider);
+      const poolId = await pool.getPoolId();
+
+      const initJoinParams = weightedPoolFactory.buildInitJoin({
+        joiner: signerAddress,
+        poolId,
+        poolAddress,
+        tokensIn: tokenAddresses,
+        amountsIn: [
+          parseFixed('2000', 6).toString(),
+          parseFixed('8000', 6).toString(),
+        ],
+      });
+
+      const iERC20 = [
+        'function approve(address,uint256) nonpayable',
+        'function balanceOf(address) view returns(uint)',
+      ];
+      const erc20 = new Contract(AddressZero, iERC20);
+      // Approve vault for seeder
```

Comment:
You don't need to do approvals here. `forkSetup` already does that.
This was causing your issues with mainnet. If you remove it you can also switch to mainnet and remove goerli config.

Reply:
Done! Thank you, didn't know forkSetup did that.